sync/firewall/firewall.go

```diff
@@ -197,6 +203,47 @@ func (*Firewall) getIPFromMultiAddress(address string) (string, error) {
 	return ip, nil
 }
 
+func (f *Firewall) isDuplicatePeer(pid peer.ID) bool {
+	p := f.peerSet.GetPeer(pid)
```

Comment:
We have observed that LibP2P sometimes misses the disconnect event. Therefore, this may cause nodes to get banned even if they run on one network ID.

Reply:
Do you explain more details?